src/include/modify_inline.h

```diff
@@ -0,0 +1,111 @@
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#pragma once
+
+#define WT_MODIFY_FOREACH_BEGIN(mod, p, nentries, napplied)                    \
+    do {                                                                       \
+        const size_t *__p = p;                                                 \
+        const uint8_t *__data = (const uint8_t *)(__p + (size_t)(nentries)*3); \
+        int __i;                                                               \
+        for (__i = 0; __i < (nentries); ++__i) {                               \
+            memcpy(&(mod).data.size, __p++, sizeof(size_t));                   \
+            memcpy(&(mod).offset, __p++, sizeof(size_t));                      \
+            memcpy(&(mod).size, __p++, sizeof(size_t));                        \
+            (mod).data.data = __data;                                          \
+            __data += (mod).data.size;                                         \
+            if (__i < (napplied))                                              \
+                continue;
+
+#define WT_MODIFY_FOREACH_REVERSE(mod, p, nentries, napplied, datasz) \
+    do {                                                              \
+        const size_t *__p = (p) + (size_t)(nentries)*3;               \
+        const uint8_t *__data = (const uint8_t *)__p + datasz;        \
+        int __i;                                                      \
+        for (__i = (napplied); __i < (nentries); ++__i) {             \
+            memcpy(&(mod).size, --__p, sizeof(size_t));               \
+            memcpy(&(mod).offset, --__p, sizeof(size_t));             \
+            memcpy(&(mod).data.size, --__p, sizeof(size_t));          \
+            (mod).data.data = (__data -= (mod).data.size);
+
+#define WT_MODIFY_FOREACH_END \
+    }                         \
+    }                         \
+    while (0)
+
+/*
+ * __wt_modify_max_memsize --
+ *     Calculate the maximum memory usage when applying the modify.
+ */
+static WT_INLINE void
+__wt_modify_max_memsize(const void *modify, size_t *max_memsize)
+{
+    WT_MODIFY mod;
+    size_t tmp;
+    const size_t *p;
+    int nentries;
+
+    /* Get the number of modify entries. */
+    p = (const size_t *)modify;
+    memcpy(&tmp, p++, sizeof(size_t));
+    nentries = (int)tmp;
+
+    WT_MODIFY_FOREACH_BEGIN (mod, p, nentries, 0) {
+        *max_memsize = WT_MAX(*max_memsize, mod.offset) + mod.data.size;
+    }
+    WT_MODIFY_FOREACH_END;
+}
+
+/*
+ * __wt_modify_max_memsize_format --
+ *     Calculate the maximum memory usage when applying the modify.
+ */
+static WT_INLINE void
+__wt_modify_max_memsize_format(const void *modify, const char *value_format, size_t *max_memsize)
+{
+    __wt_modify_max_memsize(modify, max_memsize);
+
+    if (value_format[0] == 'S')
+        ++(*max_memsize);
+}
+
+/*
+ * __wt_modify_max_memsize_unpacked --
+ *     Calculate the maximum memory usage when applying the modify.
```

Comment:
Can we update the function descriptions so they are not the same?

Reply:
[06af2b3](https://github.com/wiredtiger/wiredtiger/pull/10739/commits/06af2b339e2e6bd40202108b5b15c1a8d1b5af14)